packages/python/plotly/plotly/express/_imshow.py

```diff
@@ -403,6 +397,27 @@ def imshow(
                 _vectorize_zvalue(zmin, mode="min"),
                 _vectorize_zvalue(zmax, mode="max"),
             )
+        x0, y0, dx, dy = (None,) * 4
+        if x is not None:
+            x = np.asanyarray(x)
+            if np.issubdtype(x.dtype, np.number):
+                x0 = x[0]
+                dx = x[1] - x[0]
+            else:
+                raise ValueError(
+                    "Only numerical values are accepted for the `x` parameter "
```

Comment:
this is correct but I think a bit opaque for users? in the xarray case they've maybe not specified `x` (as it's implicit) and they don't know they're using an `Image` trace (all they know is they set `binary_string=True`

Reply:
ok I can think of a better error message, but do we agree that we should also error when `x` is passed explicitly and does not have a numerical dtype? (otherwise it would just be ignored, which I agree is not a good thing).

Reply:
I changed the error message, is it better now?